src/server/pfs/server/driver.go

```diff
@@ -1809,6 +1283,201 @@ func (d *driver) clearCommit(ctx context.Context, commit *pfs.Commit) error {
 	return errors.EnsureStack(d.commitStore.DropFileSets(ctx, commit))
 }
 
+func (d *driver) fillNewBranches(txnCtx *txncontext.TransactionContext, branch *pfs.Branch, provenance []*pfs.Branch) error {
+	repoBranches := map[*pfs.Repo][]*pfs.Branch{branch.Repo: {branch}}
+	newRepoCommits := make(map[string]*pfs.Commit)
```

Comment:
can this be simplified now that a DAG cannot contain multiple branches from the same repo?